ls1intum/artemis-android
# `Chore`: Migrate to coil3 + use ArtemisImageProvider

### Problem Description
See #163 

### Changes
 
- Updated dependencies to use coil3
  - Except for the ImageLoader used for markwon, as it does not support coil3. This should be addressed separately in #171 
- Using `ArtemisImageProvider` for 
  - CourseIcons -> Deleted `CourseImageProvider`
  - ImageLoader used in Markwon
  - Drag n Drop exercises

### Steps for testing
Go to screens with images and see that everything works as expected
- CourseOverview -> course icons
- RegisterForCourse -> course icons (eg testUser1 on TS1 has a course he can enroll to)
- ChatList -> posts show profilie picture
- ChatList -> posts with images are displayed
- InstanceSelection -> server icons 


- Drag n drop quiz currently can not be tested -> Added as todo to #107
